### ISSUE TITLE
Links handler in custom reports

### ISSUE THREAD

**Issue by vomikan** [moneymanagerex/moneymanagerex#2746]
MMEX version:
 - [x] 1.3.6
 - [x] 1.3.5
 - [ ] 1.3.4 or older
 - [ ] 1.4.X

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

Attachments were not shown in reports when using

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Related issue #2744

**Comment by vomikan** [comment]
That my report. Just for check how links working.
[Attachments.grm.zip](https://github.com/moneymanagerex/moneymanagerex/files/5698721/Attachments.grm.zip)

**Comment by vomikan** [comment]
@TomAff
your report contains the "file://" prefix for files links. I've changed the code to support this kind of links.

```select Status, strftime ('%d.%m.%Y', date) as Datum
, payee as Empfaenger, notes as Buchungstext
, round(amount*-1,2)  Betrag
, attDocument as Beleg
, 'file://D:\Software\MoneyManagerEx\MMEX_privat_Attachments\Transaction\'||attFilename as Belegdatei,
(select round(sum(amount)*-1,2) from alldata  where subcategory='2012 - Anbau') as summe
from alldata  where subcategory='2012 - Anbau' order by date, id
```

**Comment by TomAff** [comment]
Thanks a lot!

**Comment by vomikan** [comment]
new files would be avaiable soon

**Comment by vomikan** [comment]
with p as (
select 'file://D:\Software\MoneyManagerEx\MMEX_privat_Attachments\Transaction\' path
, 'attachment:Transaction|' link
),
c as (
    select * from (
	select c.CATEGNAME, c.CATEGID, s.SUBCATEGID, s.SUBCATEGNAME from CATEGORY_V1 c
	left join	SUBCATEGORY_V1 s on s.CATEGID=c.CATEGID
	union all
    select c.CATEGNAME, c.CATEGID, null, '' from CATEGORY_V1 c		
	order by 1
	) --where SUBCATEGNAME='test'
)
select p.path||a.FILENAME full_path
, p.link||t.TRANSID simple_link
, a.FILENAME
, (select ACCOUNTNAME from ACCOUNTLIST_V1 where ACCOUNTID=t.ACCOUNTID) account
, t.TRANSDATE
, t.TRANSCODE
, case c.SUBCATEGID when -1 then c.CATEGNAME else c.CATEGNAME||':'||c.SUBCATEGNAME end Category
, (select PAYEENAME from PAYEE_V1 where PAYEEID=t.PAYEEID) PAYEENAME
, t.TRANSAMOUNT
from ATTACHMENT_V1 a
inner join CHECKINGACCOUNT_V1 t  on a.REFID=t.TRANSID
inner join c on t.CATEGID=c.categid and t.SUBCATEGID=c.SUBCATEGID
left join p

**Comment by vomikan** [comment]
One more issue.
![image](https://user-images.githubusercontent.com/6836805/102711762-2f0c5b80-42cd-11eb-9f2c-a10b67ca6f19.png)
In case if https:// prefix is missing 
![image](https://user-images.githubusercontent.com/6836805/102711782-4d725700-42cd-11eb-9fa9-593516afbe69.png) you can't enter it but this value may be already there before upgrade.

![image](https://user-images.githubusercontent.com/6836805/102711826-814d7c80-42cd-11eb-886e-2fad6f57e86d.png)

**Comment by vomikan** [comment]
```^(https?:\/\/)?([0-9a-z_\.]+)\.([a-z]{2,6}\.?)(\/[0-9a-z_\.]*)*\/?$```
https://regex101.com/

**Comment by TomAff** [comment]
File-Links still does not work?!
<a href="<TMPL_VAR full_path>"><TMPL_VAR full_path></a> 
<a href=<TMPL_VAR full_path>><TMPL_VAR full_path></a> 
<a href="<TMPL_VAR full_path>" target="_bank"><TMPL_VAR full_path></a> 
<a href=<TMPL_VAR full_path> target="_bank"><TMPL_VAR full_path></a> 

GRM-File [ATT-Test.zip](https://github.com/moneymanagerex/moneymanagerex/files/6643558/ATT-Test.zip)

**Comment by vomikan** [comment]
@TomAff Everythin is fine.
[Attachments2.grm.zip](https://github.com/moneymanagerex/moneymanagerex/files/6643901/Attachments2.grm.zip)
The "Transaction" folder added after Attachments.
Your code:
```
select 'file://C:\Dev\mmex\MMEX_recent_7_Attachments\' path
```
My:
```
select 'file://C:\temp\MMEX_CPU_Attachments\Transaction\' path

```

**Comment by TomAff** [comment]
I tried it. It didn't work.

Tested this way too
I copy a pdf to D:\123.pdf

SQL
select 'file://D:\123.pdf' test, p.path||a.FILENAME full_path ....

Template
          <td><a href="<TMPL_VAR "test">" target='_blank'><TMPL_VAR "TEST"></a></td>
    
Nothing happens... (Save target as works)

simple link works.

**Comment by vomikan** [comment]
Seems the line incorrect:
<a href="<TMPL_VAR "test">" 
Change it to:
<a href="<TMPL_VAR 'test'>" 
Or
<a href="<TMPL_VAR test>"

**Comment by TomAff** [comment]
I tried these 
        <td><a href="<TMPL_VAR 'test'>"><TMPL_VAR "TEST"></a></td>
        <td><a href="<TMPL_VAR test>"><TMPL_VAR "TEST"></a></td>
        <td><a href="<TMPL_VAR 'test'>" target="_blank"><TMPL_VAR "TEST"></a></td>
        <td><a href="<TMPL_VAR test>" target="_blank"><TMPL_VAR "TEST"></a></td>

Save as works, but it did'nt open

**Comment by vomikan** [comment]
There is a right click menu with Options available:

![image](https://user-images.githubusercontent.com/6836805/121810565-18d2cd00-cc6a-11eb-8f9f-ab287ec074fc.png)

It may be copied then pasted to the Run (Win+R) window:
![image](https://user-images.githubusercontent.com/6836805/121810629-4a4b9880-cc6a-11eb-8d2c-8286f54bf3c5.png)

**Comment by TomAff** [comment]
That works. PDF opens.
![2021-06-13 16_13_02-Ausführen](https://user-images.githubusercontent.com/46572503/121810754-5895b680-cc62-11eb-8440-a1e74593f24c.jpg)

**Comment by vomikan** [comment]
Hmmm. Seems everything OK. (But it should be file://D:/123.pdf)

**Comment by TomAff** [comment]
I tried it without SQL

      `  <td><a href="file://D:/123.pdf"><TMPL_VAR "TEST"></a></td>`
      
Link is always shown in properties as
file:///D:/123.pdf

even if I use 
      `  <td><a href="D:/123.pdf"><TMPL_VAR "TEST"></a></td>`

file:/// is alway set before

**Comment by vomikan** [comment]
I've added to the Template tab:

```  <td><a href="D:/123.pdf" target="_blank">TEST</a></td>```
![image](https://user-images.githubusercontent.com/6836805/121811982-d95aaf80-cc6e-11eb-88b5-83f048191b36.png)
Working

**Comment by TomAff** [comment]
`<td><a href="D:/123.pdf" target="_blank">TEST</a></td>`

Added => doesn't work.

Is it possible that there is a dll or something missing, that you have in a folder, that you have set as "Path"

**Comment by TomAff** [comment]
A short time later this message comes up
![2021-06-13 16_50_59-MoneyManagerEx Error](https://user-images.githubusercontent.com/46572503/121812325-b8db2700-cc67-11eb-991c-e1f7a857ac55.jpg)

**Comment by vomikan** [comment]
I've got the same error when trying to zoom (Ctrl + mouse wheel) home page. This is a separate issue.
